### PR TITLE
Make `DriverClientProxy` work with `RecordSet` and `task_type`.

### PR DESCRIPTION
**Pull request by panh99**
- Updated `DriverClientProxy` so that it converts/retrieves legacy `*Ins`/`*Res` to/from `RecordSet` and `task_type`.
- Updated the unit tests accordingly.

This PR is based on #2842.

### CODE REVIEW

File: src/py/flwr/server/utils/validator.py [hunk 1]
```diff
@@ -64,21 +64,8 @@ def validate_task_ins_or_res(tasks_ins_res: Union[TaskIns, TaskRes]) -> List[str
...
-            err_msg = ", ".join([f"`{field}`" for field in has_fields])
-            validation_errors.append(
-                f"`task` in `TaskIns` must set at least one of fields {{{err_msg}}}"
-            )
-        if has_fields[
-            "legacy_server_message"
-        ] and not tasks_ins_res.task.legacy_server_message.HasField("msg"):
-            validation_errors.append("`legacy_server_message` does not set field `msg`")
+        if not tasks_ins_res.task.HasField("recordset"):
+            validation_errors.append("`recordset` MUST be set")
```
Comment (danieljanes) [inline src/py/flwr/server/utils/validator.py:68]: Should we add `task_type` as mandatory as well?

File: src/py/flwr/server/utils/validator.py [hunk 2]
```diff
@@ -115,21 +102,8 @@ def validate_task_ins_or_res(tasks_ins_res: Union[TaskIns, TaskRes]) -> List[str
...
-            err_msg = ", ".join([f"`{field}`" for field in has_fields])
-            validation_errors.append(
-                f"`task` in `TaskRes` must set at least one of fields {{{err_msg}}}"
-            )
-        if has_fields[
-            "legacy_client_message"
-        ] and not tasks_ins_res.task.legacy_client_message.HasField("msg"):
-            validation_errors.append("`legacy_client_message` does not set field `msg`")
+        if not tasks_ins_res.task.HasField("recordset"):
+            validation_errors.append("`recordset` MUST be set")
```
Comment (danieljanes) [inline src/py/flwr/server/utils/validator.py:106]: Should we add `task_type` as mandatory as well?

File: src/py/flwr/server/utils/validator_test.py [hunk 3]
```diff
@@ -148,12 +94,7 @@ def create_task_ins(
             delivered_at=delivered_at,
             producer=Node(node_id=0, anonymous=True),
             consumer=consumer,
-            legacy_server_message=ServerMessage(
-                reconnect_ins=ServerMessage.ReconnectIns()
-            )
-            if has_legacy_server_message
-            else None,
-            sa=SecureAggregation(named_values={}) if has_sa else None,
+            recordset=RecordSet(parameters={}, metrics={}, configs={}),
```
Comment (danieljanes) [inline src/py/flwr/server/utils/validator_test.py:97]: What about `task_type`?

File: src/py/flwr/driver/driver_client_proxy_test.py [hunk 4]
```diff
@@ -65,9 +97,9 @@ def test_get_properties(self) -> None:
                         group_id="",
                         run_id=0,
                         task=task_pb2.Task(  # pylint: disable=E1101
-                            legacy_client_message=ClientMessage(
-                                get_properties_res=ClientMessage.GetPropertiesRes(
-                                    properties=CLIENT_PROPERTIES
+                            recordset=_make_recordset_proto(
```
Comment (danieljanes) [inline src/py/flwr/driver/driver_client_proxy_test.py:100]: Should we add `task_type` to all of these test cases as well?

File: src/py/flwr/server/utils/validator.py (hunk 1 above)
Comment (panh99) [reply src/py/flwr/server/utils/validator.py:68]: I think checking `task_type` makes sense. Will update this.

**Review by danieljanes — APPROVED** [review]

**Review by danieljanes — APPROVED** [review]

**Pull request merged**